src/streaming/text/TextTracks.js

```diff
@@ -493,16 +495,22 @@ function TextTracks(config) {
         }
 
         for (let item = 0; item < captionData.length; item++) {
-            let cue;
+            let cue = null;
             const currentItem = captionData[item];
 
             track.cellResolution = currentItem.cellResolution;
             track.isFromCEA608 = currentItem.isFromCEA608;
 
             if (!isNaN(currentItem.start) && !isNaN(currentItem.end)) {
-                cue = currentItem.type === 'html' && captionContainer ? _handleHtmlCaption(currentItem, timeOffset, track)
-                    : currentItem.data ? _handleNonHtmlCaption(currentItem, timeOffset, track) : null;
+                if (dispatchForManualRendering) {
+                    cue = _handleCaptionEvents(currentItem, timeOffset, track, item);
```

Comment:
When playing TTML subtitles in the reference UI with `dispatchForManualRendering= true`the rendering is broken. Switching between subtitles causes "undefined" to be rendered. Moreover,  `_fireCueEnter` does not provide all the necessary information to do a custom rendering of TTML subtitles.

Can you please check if you can add the logic of `_handleCaptionEvents` to the `_handleHtmlCaption` and `_handleNonHtmlCaption` functions. Basically defining different `cue.onenter` and ` cue.onexit` functions depending on whether  `dispatchForManualRendering` is enabled.

An even cleaner approach might be to introduce a function `_getCueInformation` and return the required fields depending on whether the cue is an HTML oder non-HTML cue. Then you can probably keep the logic you implemented here and have all the right information to also render HTML cues. Still we need to fix the error in the reference UI with rendering `undefined`. As an example you can use this MPD: https://livesim2.dashif.org/vod/testpic_2s/multi_subs.mpd